### DISCUSSION TITLE
Adding xcframework as dependency to native modules

### DISCUSSION THREAD

**Discussion by Kahan** [callstack/react-native-builder-bob#292] [Q&A] [answered]
I've been trying to add a native xcframework library as a dependency to my iOS native modules. The iOS root application is missing the general tab in Xcode which is where one would as the framework dependency. I was able to successfully link an .aar library to my android native modules. Am I missing something while implementing in iOS or is it not supported?

**Comment by nicklasbring** [comment]
I have the same question. I also managed to include a .aar on the Android side, but i can't figure it out on iOS...

**Comment by pulnec** [comment]
That to the friends, to add the xcframework in the .podspec file that is in the root of the project you must add the line.

s.vendored_frameworks = 'ios/exmaple.xcframework'

you must keep the xcframework inside the "ios" folder that is in the root.

**Comment by nicklasbring** [comment]
Thank you Pulnec. 
I did get it to work when adding the dependency as a pod in the root podspec though. 
First time a tried to add it as a pod in the podspec, it also installed to pod, but it seemed like the link between the my own module and the pod wasn't fully there.  The second time i tried it, it worked just fine.

**Comment by pulnec** [comment]
@nicklasbring, Excellent, can you leave an example here, of how you did it?

**Comment by TomNes94** [comment]
@nicklasbring @pulnec Did anyone of you manage to solve this? I get as far as the Native Module recognizing the xcframework. The example app can import the xcframework just fine when adding as embedded binary, but using the native module from the example gives the "Module not found" error.

**Comment by dnelsonlivecurrence** [comment, marked as answer]
## iOS

I was able to add the following to my podspec to get the native module I was working with imported:

For the XCFramework:
`s.vendored_frameworks = 'ios/MiSnapSDK/*.{xcframework}'`

For the needed resources (in this case there were multiple language resources and I had to make sure to import only English):
`s.resources = "ios/MiSnapSDK/MiSnapUX/**/*.{png,jpeg,jpg,storyboard,xib,xcassets}","ios/MiSnapSDK/MiSnapUX/UX_Resources/en.lproj/MiSnapSDKLocalizable.strings"`

And I modified the source files to only import the bridge code, and the .h / .m files that weren't imported with the vendor frameworks:
`s.source_files = "ios/*.{h,m,mm}","ios/MiSnapSDK/MiSnapUX/UX_Files/*.{h,m,mm}","ios/MiSnapSDK/MiSnapUX/UX2_Files/*.{h,m,mm}"`

When I used the default `ios/**/*.{h,m,mm}` I was getting build errors about duplicate files `Error: Multiple commands produce`

## Android

Under dependencies in the build.gradle I added the associated .aar files like so:

```
  compileOnly files('libs/api-release.aar')
  compileOnly files('libs/barcode-release.aar')
  compileOnly files('libs/barcodecontroller-release.aar')
  compileOnly files('libs/misnapcamera-release.aar')
  compileOnly files('libs/mibidata-release.aar')
  compileOnly files('libs/imageutils-release.aar')
  compileOnly files('libs/misnapscience-release.aar')
  compileOnly files('libs/misnapworkflow_UX2-release.aar')
  compileOnly files('libs/misnapcontroller-release.aar')
  compileOnly files('libs/misnaphybridcontroller-release.aar')
  compileOnly files('libs/misnapdetector-release.aar')
  compileOnly files('libs/misnapextractioncontroller-release.aar')
```

And then in the project's app/build.gradle I implemented the .aar files with `implementation files('../../node_modules/react-native-misnap/android/libs/api-release.aar')` for each file.

**Reply by atlj** [reply]
It's also important to note which files to edit:

#### iOS
`ios/your_project_name.podspec`

#### Android
`android/build.gradle`

And please also keep in mind that if you are editing `app/build.gradle`, every consumer of the library should do the same.

**Comment by ajakka** [comment]
I have a similar issue but I want the developer to obtain the xcframwork (because it's under a paywall) and import it somehow for my library to use. the xcframwork is swift only. I fixed it using the following:

on my example project I put the xcframwork along with a dummy podspec in Frameworks folder and then install it using the Podfile:

```ruby
  use_react_native!(
  ...
  )

  pod 'heresdk', :path => 'Frameworks'
```
 
This approach used to work and the swift files on my library also find the `heresdk` when I import it, but It just magically stooped working after a while. Any idea why?

Error message:

```zsh
2023-12-10 18:55:59.604 xcodebuild[10512:91934] DVTCoreDeviceEnabledState: DVTCoreDeviceEnabledState_Disabled set via user default (DVTEnableCoreDevice=disabled)
** BUILD FAILED **


The following build commands failed:
        SwiftEmitModule normal x86_64 Emitting\ module\ for\ react_native_maps_here (in target 'react-native-maps-here' from project 'Pods')
(1 failure)
```
